### ISSUE TITLE
implement midpoint derivative calculation

### ISSUE THREAD

**Issue by Swanson-Hysell** [PmagPy/PmagPy#709]
Currently within `ipmag.curie()` there is code to calculate the derivative which is then smoothed. It seems preferable to use the midpoint temperature (i.e. `T_mid = (T[i + 1] + T[i]) / 2`) for the derivative rather than the original T array values in order to associate the derivative value between the two points from which it is calculated.

Here is the current relevant code:
```
    # calculate first derivative
    d1, T_d1 = [], []
    for i in range(len(M_smooth) - 1):
        Dy = M_smooth[i - 1] - M_smooth[i + 1]
        Dx = T[i - 1] - T[i + 1]
        d1.append(Dy/Dx)
    T_d1 = T[1:len(T - 1)]
    d1 = np.array(d1, 'f')
    d1_smooth = smooth(d1, window_len)
```

**Comment by apivarunas** [comment]
@Swanson-Hysell I took a look at this issue. I've actually never used the `ipmag.curie()` function since the function initially enforces even time-steps (=1.0) or quits. And I don't know about other Kappabridge machines, but I know that neither I've used has done that. I believe the function is able to handle and interpolate irregular data, one just needs to reorder the initial statements. I've almost got it working with a test file, I just have one last bug that must be related to the length of the array since it only crops up on the last pair of values.

I'll send out a PR related to that soon-ish.

**Digging into the actual raised issue**, I think that the original T array values do in fact represent the midpoint, since the calculation of the derivative is going from `Dx = T[i - 1] - T[i + 1]`. That can be adjusted to be consecutive M/T steps and the midpoint  implemented. Would you like that to be the case?

**Comment by Swanson-Hysell** [comment]
Good point here. The current approach is calculating Dx across the step before and after the temperature. This approach actually implements a bit of smoothing by spanning more temperatures steps rather than just consecutive ones. Given that there is already a pre-smoothing step, it seems like it could make sense to modify the function to be consecutive M/T steps with the midpoint implemented. Does that approach make the most sense to you?
